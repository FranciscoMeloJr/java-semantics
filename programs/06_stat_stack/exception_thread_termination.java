// Testing that exception propagates through method calls
// The last line is not printed due to unknown java-prototype definition bug.

public class exception_thread_termination {

  public static void main(String[] args) {
    try {
      try {
        throwEx();

        System.out.println("No exception");
      } finally {
        System.out.println("finally after uncaught exception");
      }
    } finally {
      System.out.println("finally 2 after uncaught exception");
    }
  }

  void throwEx() {
    throw 5;
  }
}

// finally after uncaught exception
// finally 2 after uncaught exception
// Thread terminated with exception: 5
