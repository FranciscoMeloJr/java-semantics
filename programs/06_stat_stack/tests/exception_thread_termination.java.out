finally after uncaught exception
finally 2 after uncaught exception
Thread terminated with exception: 5
